packages/server/src/shared/internal/serialize.ts

```diff
@@ -18,18 +18,28 @@ type IsAny<T> = 0 extends T & 1 ? true : false;
 // support it as both a Primitive and a NonJsonPrimitive
 type JsonReturnable = JsonPrimitive | undefined;
 
-// prettier-ignore
-export type Serialize<T> =
-  IsAny<T> extends true ? any :
-  T extends JsonReturnable ? T :
-  T extends Map<any, any> | Set<any> ? object :
-  T extends NonJsonPrimitive ? never :
-  T extends { toJSON(): infer U } ? U :
-  T extends [] ? [] :
-  T extends [unknown, ...unknown[]] ? SerializeTuple<T> :
-  T extends readonly (infer U)[] ? (U extends NonJsonPrimitive ? null : Serialize<U>)[] :
-  T extends object ? Simplify<SerializeObject<UndefinedToOptional<T>>> :
-  never;
+/* prettier-ignore */
+export type Serialize<T> = IsAny<T> extends true
+  ? any
+  : unknown extends T
+  ? unknown
```

Comment:
Sorry about this diff. The only thing this commit actually changes about `type Serialize` is it adds explicit handling for `unknown` here